pytest_tests/testsuites/conftest.py

```diff
@@ -134,35 +135,47 @@ def check_binary_versions(request, hosting: Hosting, client_shell: Shell):
 
 @pytest.fixture(scope="session")
 @allure.title("Prepare tmp directory")
-def temp_directory():
+def temp_directory() -> str:
     with allure.step("Prepare tmp directory"):
         full_path = os.path.join(os.getcwd(), ASSETS_DIR)
-        shutil.rmtree(full_path, ignore_errors=True)
-        os.mkdir(full_path)
+        create_dir(full_path)
 
     yield full_path
 
     with allure.step("Remove tmp directory"):
-        shutil.rmtree(full_path)
+        remove_dir(full_path)
 
 
 @pytest.fixture(scope="session", autouse=True)
-@allure.title("Collect logs")
-def collect_logs(temp_directory, hosting: Hosting):
+@allure.title("Collect full logs")
+def collect_full_tests_logs(temp_directory, hosting: Hosting):
+    test_name = 'full_logs'
     start_time = datetime.utcnow()
     yield
     end_time = datetime.utcnow()
-
-    # Dump logs to temp directory (because they might be too large to keep in RAM)
     logs_dir = os.path.join(temp_directory, "logs")
-    dump_logs(hosting, logs_dir, start_time, end_time)
-    attach_logs(logs_dir)
+    store_logs(hosting, logs_dir, test_name, start_time, end_time)
     check_logs(logs_dir)
 
 
+@pytest.fixture(scope="function", autouse=True)
+@allure.title("Collect test logs")
+def collect_test_logs(request, temp_directory, hosting: Hosting):
+    test_name = request.node.nodeid.replace(":", "_").replace("[", "_").replace("]", "_").replace("/", "_")
+    hash_suffix = hashlib.md5(test_name.encode()).hexdigest()
+    file_name = (test_name[:200] + '_' + hash_suffix)  # limit total length to 255
```

Comment:
Do we need a hash if `test_name` fits into `200`?

Reply:
> Do we need a hash if `test_name` fits into `200`?

In my opinion, yes, hash is needed in this case too.
Standard use case - comparing two test runs. It's very convenient to download two archives and compare them by hash.
It seems to me that it will make life easier:)